host/contracts/manager_test.go

```diff
@@ -177,6 +175,112 @@ func TestV2ContractLifecycle(t *testing.T) {
+					Address: fc.RenterOutput.Address,
+					Value:   fc.RenterOutput.Value.Add(renterAllowance),
+				},
+				HostOutput: types.SiacoinOutput{
+					Address: fc.HostOutput.Address,
+					Value:   fc.HostOutput.Value.Add(collateral),
+				},
+				MissedHostValue: fc.MissedHostValue.Add(collateral),
+				TotalCollateral: fc.TotalCollateral.Add(collateral),
+				RenterPublicKey: renterKey.PublicKey(),
+				HostPublicKey:   hostKey.PublicKey(),
+			},
+			HostRollover:   fc.HostOutput.Value,
+			RenterRollover: fc.RenterOutput.Value,
+		}
+		renewalSigHash := cs.RenewalSigHash(renewal)
+		renewal.HostSignature = hostKey.SignHash(renewalSigHash)
+		renewal.RenterSignature = renterKey.SignHash(renewalSigHash)
+		contractSigHash := cs.ContractSigHash(renewal.NewContract)
+		renewal.NewContract.HostSignature = hostKey.SignHash(contractSigHash)
+		renewal.NewContract.RenterSignature = renterKey.SignHash(contractSigHash)
+
+		_, fce, err := con.V2FileContractElement(contractID)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		fundAmount := cs.V2FileContractTax(renewal.NewContract).Add(collateral).Add(renterAllowance)
+		renewalTxn := types.V2Transaction{
+			FileContractResolutions: []types.V2FileContractResolution{
+				{
+					Parent:     fce,
+					Resolution: &renewal,
+				},
+			},
+		}
+		basis, toSign, err := w.FundV2Transaction(&renewalTxn, fundAmount, false)
+		if err != nil {
+			t.Fatal("failed to fund transaction:", err)
+		}
+		w.SignV2Inputs(&renewalTxn, toSign)
+		return contractID.V2RenewalID(), renewal.NewContract, rhp4.TransactionSet{
+				Basis:        basis,
+				Transactions: []types.V2Transaction{renewalTxn},
+			}, proto4.Usage{
+				RiskedCollateral: renewal.NewContract.RiskedCollateral(),
+			}
+	}
+
+	appendSector := func(t *testing.T, contractID types.FileContractID, fc types.V2FileContract, corrupt bool) (types.V2FileContract, []types.Hash256, proto4.Usage) {
+		var sector [proto4.SectorSize]byte
+		frand.Read(sector[:])
+
+		var root types.Hash256
+		if corrupt {
+			root = frand.Entropy256()
+		} else {
+			root = proto4.SectorRoot(&sector)
+		}
+		roots := []types.Hash256{root}
+
+		if err := node.Volumes.Write(root, &sector); err != nil {
+			t.Fatal(err)
+		}
+
+		fc.Filesize = proto4.SectorSize
+		fc.Capacity = proto4.SectorSize
+		fc.FileMerkleRoot = proto4.MetaRoot(roots)
+		fc.RevisionNumber++
+		// transfer some funds from the renter to the host
+		cost, collateral := types.Siacoins(1), types.Siacoins(2)
+		fc.RenterOutput.Value = fc.RenterOutput.Value.Sub(cost)
+		fc.HostOutput.Value = fc.HostOutput.Value.Add(cost)
+		fc.MissedHostValue = fc.MissedHostValue.Sub(collateral)
+		sigHash := node.Chain.TipState().ContractSigHash(fc)
+		fc.HostSignature = hostKey.SignHash(sigHash)
+		fc.RenterSignature = renterKey.SignHash(sigHash)
+
+		usage := proto4.Usage{
+			Storage:          cost,
+			RiskedCollateral: collateral,
+		}
+		err := node.Contracts.ReviseV2Contract(contractID, fc, roots, usage)
```

Comment:
I'd also create an account and credit it. You can't upload without funding an account first and uploading to temp storage.